pkg/cmd/roachprod/cli/list.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/cloud"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/prometheus"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/vm"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/vm/gce"
+	"github.com/cockroachdb/cockroach/pkg/util/sparkline"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+	"github.com/fatih/color"
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+const (
+	clusterQueryTemplate = "sum(rate(txn_commits{cluster=~\"$CLUSTERS\"}[5m])) by (cluster)"
+
+	clusterHdr     = "Cluster"
+	cloudsHdr      = "Clouds"
+	sizeHdr        = "Size"
+	vmHdr          = "VM"
+	archHdr        = "Arch"
+	costPerHourHdr = "$/hour"
+	costSpentHdr   = "$ Spent"
+	costPerTTLHdr  = "$/TTL"
+	uptimeHdr      = "Uptime"
+	ttlHdr         = "TTL"
+	txnCommitsHdr  = "Txn_Commits"
+)
+
+var (
+	allHeaders = [11]string{clusterHdr, cloudsHdr, sizeHdr, vmHdr, archHdr, costPerHourHdr, costSpentHdr, costPerTTLHdr, uptimeHdr, ttlHdr, txnCommitsHdr}
+)
+
+func clusterToSparkline(clusterNames []string) (map[string]string, error) {
+	res := make(map[string]string)
+	query := strings.ReplaceAll(clusterQueryTemplate, "$CLUSTERS", strings.Join(clusterNames, "|"))
+	from := timeutil.Now().Add(time.Duration(-60) * time.Minute)
+	to := timeutil.Now()
+
+	promClient, err := prometheus.NewPromClient(context.Background(), prometheus.DefaultPromHostUrl, true)
+	if err != nil {
+		return res, err
+	}
+	m, err := prometheus.QueryRange(context.Background(), promClient, query, from, to, prometheus.DefaultScrapeInterval)
+	if err != nil {
+		return res, err
+	}
+	var errs error
+	for _, labelValToSeries := range m {
+		for clusterName, series := range labelValToSeries {
+			points := prometheus.Values(series)
+			img, err := sparkline.DrawSparkline(points, 250, 35)
+			if img != nil && err == nil {
+				if s, err := sparkline.ITermEncodePNGToString(img, ""); err != nil {
```

Comment:
Nit: Wouldn't it be better to have a `sparkline.DrawAsItermString(points, 250, 35)` that combines both calls above?